### PR TITLE
CO-3284 Pending state is now replaced by an activity

### PR DESCRIPTION
**Pull request by Theyiot**
We simply remove legacy inheritance according to other modifications from here: https://github.com/CompassionCH/compassion-modules/pull/1447

### CODE REVIEW

**Review by ecino — CHANGES_REQUESTED** [review]
Please remove the `state` field inside `partner_compassion.py` and all related usages.

**Review by ecino — CHANGES_REQUESTED** [review]
You should still replace remove the fields from the view `partner_compassion_view.xml`

**Comment by Theyiot** [comment]
> You should still replace remove the fields from the view `partner_compassion_view.xml`

Last commit was amended with all TODOs removed.

File: sponsorship_switzerland/models/contracts.py [hunk 1]
```diff
@@ -193,9 +193,7 @@ def contract_active(self):
         Update partner to add the 'Sponsor' category
         """
         super().contract_active()
-        # TODO CO-3284 Replace this check by looking if mail.activity exists
-        need_validation = self.filtered(lambda s: s.partner_id.state != "active")
-        if need_validation:
+        if self.partner_id.activity_ids or self.correspondent_id.activity_ids:
```
Comment (ecino) [inline sponsorship_switzerland/models/contracts.py]: The method may be multi (self is a recordset) therefore it's forbidden to use `self.partner_id` or `self.correspondent_id`.
You can use the `mapped` method to circumvent this problem.
```suggestion
        if self.mapped("partner_id.activity_ids") or self.mapped("correspondent_id.activity_ids"):
```

**Review by ecino — CHANGES_REQUESTED** [review]

File: sponsorship_switzerland/models/contracts.py (hunk 1 above)
Comment (Theyiot) [reply sponsorship_switzerland/models/contracts.py]: I (wrongly) assumed these methods would be called on singleton recordset. This is fixed by next commit.

**Review by ecino — APPROVED** [review]

**Pull request merged**